pandas/core/strings/accessor.py

```diff
@@ -1683,19 +1683,23 @@ def zfill(self, width):
 
         Note that ``10`` and ``NaN`` are not strings, therefore they are
         converted to ``NaN``. The minus sign in ``'-1'`` is treated as a
-        regular character and the zero is added to the left of it
+        special character and the zero is added to the right of it
         (:meth:`str.zfill` would have moved it to the left). ``1000``
         remains unchanged as it is longer than `width`.
 
         >>> s.str.zfill(3)
-        0     0-1
+        0     -01
         1     001
         2    1000
         3     NaN
         4     NaN
         dtype: object
         """
-        result = self.pad(width, side="left", fillchar="0")
```

Comment:
Shouldn't this just call `self.zfill`?

Reply:
It shouldn't call itself, otherwise, it will lead to stack overflow. Also, I didn't find any other implemented `zfill` method for this object.

Reply:
Ah sorry, you're correct. I was looking at the wrong code.